pkg/kubelet/preemption/preemption.go

```diff
@@ -199,12 +200,26 @@ func (a admissionRequirementList) distance(pod *v1.Pod) float64 {
 
 // subtract returns a new admissionRequirementList containing remaining requirements if the provided pod
 // were to be preempted
-func (a admissionRequirementList) subtract(pods ...*v1.Pod) admissionRequirementList {
+func (a admissionRequirementList) subtract(cache map[string]int64, pods ...*v1.Pod) admissionRequirementList {
 	newList := []*admissionRequirement{}
 	for _, req := range a {
 		newQuantity := req.quantity
 		for _, pod := range pods {
-			newQuantity -= resource.GetResourceRequest(pod, req.resourceName)
```

Comment:
Any risk of the resource request update being updated after the entry is in the cache and the code using a stale value as a result? What would be the ramifications of that?

Reply:
In the current code, there is no indication that resource request being updated would invalidate the request used in previous iteration of the calculation.

I don't see locking in the func's touched by this PR.
So I assume for getPodsToPreempt, stale value can be tolerated.